impeller/entity/shaders/glyph_atlas.frag

```diff
@@ -4,20 +4,31 @@
 
 uniform sampler2D glyph_atlas_sampler;
 
+uniform FragInfo {
+  vec2 atlas_size;
+  vec4 text_color;
+  float font_has_color;
+} frag_info;
+
 in vec2 v_unit_vertex;
 in vec2 v_atlas_position;
 in vec2 v_atlas_glyph_size;
-in vec2 v_atlas_size;
-in vec4 v_text_color;
 
 out vec4 frag_color;
 
 void main() {
-  vec2 scale_perspective = v_atlas_glyph_size / v_atlas_size;
-  vec2 offset = v_atlas_position / v_atlas_size;
+  vec2 scale_perspective = v_atlas_glyph_size / frag_info.atlas_size;
+  vec2 offset = v_atlas_position / frag_info.atlas_size;
 
-  frag_color = texture(
-    glyph_atlas_sampler,
-    v_unit_vertex * scale_perspective + offset
-  ).aaaa * v_text_color;
+  if (frag_info.font_has_color == 1.0) {
```

Comment:
I think this still might not be right, because we can combine COLR and other fonts into a single atlas. When I made this unconditional I noticed that several fonts that were previously specified as white were rendered as black instead in the wonders app